src/ess/livedata/dashboard/config_store.py

```diff
@@ -68,3 +76,262 @@ def _evict_oldest(self) -> None:
+        num_to_remove = max(1, int(len(self.data) * self._cleanup_fraction))
+        # Dict maintains insertion order in Python 3.7+
+        oldest_keys = list(self.data.keys())[:num_to_remove]
+        for key in oldest_keys:
+            del self.data[key]
+
+    def _load_from_file(self) -> None:
+        """Load configurations from YAML file with file locking."""
+        if not self._file_path.exists():
+            logger.debug(
+                "Config file %s does not exist, starting empty", self._file_path
+            )
+            return
+
+        try:
+            with open(self._file_path) as f:
+                # Acquire shared lock for reading
+                fcntl.flock(f.fileno(), fcntl.LOCK_SH)
+                try:
+                    raw_data = yaml.safe_load(f)
+                    if raw_data is None:
+                        logger.debug("Config file %s is empty", self._file_path)
+                        return
+
+                    # Deserialize WorkflowId keys from strings
+                    for key_str, value in raw_data.items():
+                        try:
+                            workflow_id = WorkflowId.from_string(key_str)
+                            self.data[workflow_id] = value
+                        except (ValueError, KeyError) as e:
+                            logger.warning(
+                                "Skipping invalid config entry '%s': %s", key_str, e
+                            )
+                finally:
+                    fcntl.flock(f.fileno(), fcntl.LOCK_UN)
+
+        except yaml.YAMLError as e:
+            logger.error("Failed to parse config file %s: %s", self._file_path, e)
+            logger.warning("Starting with empty config store")
+        except Exception as e:
+            logger.error(
+                "Unexpected error loading config file %s: %s", self._file_path, e
+            )
+            logger.warning("Starting with empty config store")
+
+    def _save_to_file(self) -> None:
+        """Save configurations to YAML file with file locking."""
+        try:
+            # Serialize WorkflowId keys to strings for YAML
+            serialized_data = {str(key): value for key, value in self.data.items()}
+
+            # Write atomically: write to temp file, then rename
+            temp_path = self._file_path.with_suffix('.tmp')
+            with open(temp_path, 'w') as f:
+                # Acquire exclusive lock for writing
+                fcntl.flock(f.fileno(), fcntl.LOCK_EX)
+                try:
+                    yaml.safe_dump(
+                        serialized_data,
+                        f,
+                        default_flow_style=False,
+                        sort_keys=False,  # Preserve insertion order for LRU
+                    )
+                finally:
+                    fcntl.flock(f.fileno(), fcntl.LOCK_UN)
+
+            # Atomic rename
+            temp_path.replace(self._file_path)
```

Comment:
This happens after the lock has been released. So it is no synced with access to the persistent file in `_load_from_file`. (File locks are advisory, the OS will not enforce them.) And it is also not synced with access to the temporary file if multiple writes happen concurrently.